### PR TITLE
docs: Adding Documentation for Open Recent Menu-item

### PR DESCRIPTION
**Pull request by bhutraraghav**
docs: Solving issue #22018, adapting the additions made by PR #11166, i.e. adding documentation for adding menu-item for opening recent files.

Notes: none.

### CODE REVIEW

File: docs/tutorial/recent-documents.md [hunk 1]
```diff
@@ -27,6 +27,28 @@ const { app } = require('electron')
 app.clearRecentDocuments()
 ```
 
+
+__Adding Recent documents in Menuitem (For Mac):__
```
[the remark this answers is not in the document]
Comment (bhutraraghav) [reply docs/tutorial/recent-documents.md:31]: Yes, the open recent menu item is only available for Mac. You can read it in special roles available for Mac at https://www.electronjs.org/docs/api/menu-item . Also PR #11166 (which added this feature) also states that.

[the remark this answers is not in the document]
Comment (erickzhao) [reply docs/tutorial/recent-documents.md:31]: Yep, misread here. Thanks for the clarification!

File: docs/tutorial/recent-documents.md [hunk 2]
```diff
@@ -27,6 +27,32 @@ const { app } = require('electron')
...
```
Comment (erickzhao) [inline docs/tutorial/recent-documents.md:31]: I would suggest putting this under "macOS notes".

File: docs/tutorial/recent-documents.md [hunk 3]
```diff
@@ -27,6 +27,32 @@ const { app } = require('electron')
...
+
+![menu item Files][menu-item-image]
```
Comment (erickzhao) [inline docs/tutorial/recent-documents.md:33]:
```suggestion
![macOS Recent Documents menu item][menu-item-image]
```

File: docs/tutorial/recent-documents.md [hunk 4]
```diff
@@ -27,6 +27,32 @@ const { app } = require('electron')
...
+
+You can add recent documents in the menu item along with the clear recent submenu by adding the the following code to the menu array.
```
Comment (erickzhao) [inline docs/tutorial/recent-documents.md:35]:
```suggestion
You can add menu items to access and clear recent documents by adding the following code snippet to your menu's template.
```

File: docs/tutorial/recent-documents.md (hunk 2 above)
Comment (erickzhao) [inline docs/tutorial/recent-documents.md:31]:
```suggestion
__Adding the Recent Documents list to the application menu:__
```

Comment (bhutraraghav) [reply docs/tutorial/recent-documents.md]: Yes, That's a better idea!!

File: docs/tutorial/recent-documents.md [hunk 5]
```diff
@@ -27,6 +27,7 @@ const { app } = require('electron')
 app.clearRecentDocuments()
 ```
 
+
```
Comment (erickzhao) [inline docs/tutorial/recent-documents.md:30]: Stray newline

File: docs/tutorial/recent-documents.md [hunk 6]
```diff
@@ -39,11 +39,36 @@ will be started with the path of the file added as a command line argument.
...
+               "label":"Clear Recent",
+               "role":"clearrecentdocuments"
+            }
+         ]
+      }
+   ]
+}
+```
+
+__When a file is requested from the recent documents menu, the `open-file` event
```
Comment (codebytere) [inline docs/tutorial/recent-documents.md:66]: This should just be a normal sentence, with no heavy underscores: `__`

File: docs/tutorial/recent-documents.md [hunk 7]
```diff
@@ -39,11 +39,36 @@ will be started with the path of the file added as a command line argument.
 
 ## macOS Notes
 
-When a file is requested from the recent documents menu, the `open-file` event
-of `app` module will be emitted for it.
+__Adding the Recent Documents list to the application menu:__
```
Comment (codebytere) [inline docs/tutorial/recent-documents.md:42]: This can be an `h3`:

`### Adding the Recent Documents list to an Application Menu:`

**Comment by codebytere** [comment]
@bhutraraghav can you push an empty commit or force push to fix the lint check?

**Comment by bhutraraghav** [comment]
> @bhutraraghav can you push an empty commit or force push to fix the lint check?

Tried both but didn't work!

File: docs/tutorial/recent-documents.md [hunk 8]
```diff
@@ -39,6 +39,30 @@ will be started with the path of the file added as a command line argument.
 
 ## macOS Notes
 
+### Adding the Recent Documents list to the application menu:
+
+![macOS Recent Documents menu item][menu-item-image]
+
+You can add menu items to access and clear recent documents by adding the following code snippet to your menu's template.
+
+```javascript
```
Comment (codebytere) [inline docs/tutorial/recent-documents.md:48]: @bhutraraghav this should be `json`

Comment (bhutraraghav) [reply docs/tutorial/recent-documents.md:48]: @codebytere They were probably added by @erickzhao .

**Comment by bhutraraghav** [comment]
@codebytere removed the extra brace still it shows  "Merging is blocked- The base branch restricts merging to authorized users."
Does it clarifies something?

**Pull request merged**